### PR TITLE
fix: use iterator to load dumped data

### PR DESCRIPTION
**Pull request by numb3r3**
- use an `iterator` to fetch data from a large-scale dumped file
- remove `self._vecs` to reduce the memory usages, so as to support large index
- remove `fill_embedding` request handler as `self._vecs` is removed

### CODE REVIEW

File: jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ class FaissSearcher(Executor):
         The data will only be loaded if `requires_training` is set to True.
     :param max_num_training_points: Optional argument to consider only a subset of training points to training data from `train_filepath`.
         The points will be selected randomly from the available points
+    :param prefech_size: the number of data to pre-load into RAM memory
```
Comment (cristianmtr) [inline jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py:30]:
```suggestion
    :param prefech_size: the number of data to pre-load into RAM
```

File: jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py [hunk 2]
```diff
@@ -88,13 +91,25 @@ def __init__(
...
+            )
+            self._ids = np.array(list(ids_iter))
             self._ext2int = {v: i for i, v in enumerate(self._ids)}
-            self._vecs = np.array(list(vecs))
-            self.num_dim = self._vecs.shape[1]
-            self.dtype = self._vecs.dtype
-            self.index = self._build_index(self._vecs)
+
+            self._prefech_data = []
+            if self.prefech_size and self.prefech_size > 0:
```
Comment (cristianmtr) [inline jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py:101]: So if you have a dump of size N and prefetch of size X (X < N), how do you build the entire N index? It seems here that you're limiting to just X

Comment (cristianmtr) [reply jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py:101]: ah no, nvm, got it

File: jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py [hunk 3]
```diff
@@ -62,6 +63,7 @@ def __init__(
         normalize: bool = False,
         nprobe: int = 1,
         dump_path: Optional[str] = None,
+        prefech_size: Optional[int] = None,
```
Comment (cristianmtr) [inline jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py:66]: This should be a non-zero int by default, right? `None` doesn't help

File: jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py (hunk 2 above)
Comment (numb3r3) [reply jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py:101]: This is only the first prefetch step. The full fetch procedure happens in `self._build_partial_index` function

File: jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py (hunk 3 above)
Comment (numb3r3) [reply jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py:66]: here, `None` means load the full data at once. Yes, `None` will disable prefetch.

**Review by JoanFM — APPROVED** [review]

**Comment by cristianmtr** [comment]
Don't merge until https://github.com/jina-ai/executors/pull/72

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**